### DISCUSSION TITLE
Where can I find documentation on coverage file format?

### DISCUSSION THREAD

**Discussion by bdw429s** [ryanluker/vscode-coverage-gutters#313] [Q&A]
I'd like to build support in my [test framework](https://testbox.ortusbooks.com/) to generate test coverage file in a format that this extension can consume.  However, after reading through the docs and examples, I can't find any clear spec on the actual XML file and what it needs to contain.  I found a couple examples in the example folder, but they appeared to differ a lot between languages.

**Comment by ryanluker** [comment]
@bdw429s Thanks for opening up a discussion. Cool testing framework you have there!

The extension can consume a variety of coverage file types but we don't set these file structure specifications, instead we use a variety of parsing libraries that actually take the coverage files and return coverage data structures. https://github.com/ryanluker/vscode-coverage-gutters/blob/2c2e2d3febecccfb9244ef81134501fdb0432692/%40types/lcov-parse/index.d.ts#L4-L46

If you are looking to choose from the supported list (lcov, cobertura, clover, jacoco, etc) I would definitely go with lcov as it appears to be the most popular. I have included some documentation below that should help with generating this type of coverage.

### Library used to consume lcov files
https://www.npmjs.com/package/lcov-parse

### Documentation around the file structure
https://github.com/mitchhentges/lcov-rs/wiki/File-format#lcov-info-file-output-of-lcov-rs
http://ltp.sourceforge.net/coverage/lcov/geninfo.1.php
http://ltp.sourceforge.net/coverage/lcov/lcov.1.php

**Reply by bdw429s** [reply]
Thanks for the information.  I'll take a look at those docs and see what it takes to marshal our data into the lcov format.  We're already doing a Sonarcube format, which is XML.

**Reply by ryanluker** [reply]
@bdw429s You could check that maybe the sonarcube format might be one of the xml ones I mentioned above (I think clover is xml based?)

**Comment by ssprojecttest** [comment]
I install Coverage gutters and set up the setting to line coverage. how can I execute this in dotnet, bash terminal. When I run the command I get 
..../c/Demo
$ npm install lcov-parse
bash: npm: command not found

**Reply by ryanluker** [reply]
@ssprojecttest you will need to follow the dotnet setup instructions (see link below)
You can also create a new ticket as well if that doesn't solve the issue as this ticket is more around how to integrate with the extension.

https://github.com/ryanluker/vscode-coverage-gutters/tree/master/example/dotnet
